mysql-test/suite/innodb/t/innodb_stats.test

```diff
@@ -63,3 +63,16 @@ CREATE TABLE test_innodb_stats (
 -- disable_query_log
 DROP TABLE test_innodb_stats;
 set @@use_stat_tables= @save_use_stat_tables;
+--enable_query_log
+--echo #
+--echo #  MDEV-31740 InnoDB statistics column length validation failed
+--echo #
+call mtr.add_suppression("InnoDB: Table mysql.innodb_table_stats has length mismatch in the column name last_update. Please either run 'mariadb-upgrade' or fix it manually with: ALTER TABLE mysql.innodb_table_stats MODIFY last_update");
+SET sql_mode='maxdb';
+ALTER TABLE mysql.innodb_table_stats MODIFY last_update TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP() ON UPDATE CURRENT_TIMESTAMP();
+CREATE TABLE t (a INT KEY)Engine=InnoDB STATS_PERSISTENT=1;
+SET sql_mode=DEFAULT;
```

Comment:
Are the `sql_mode` changes really necessary? If yes, would the following work (2 statements instead of 4)?
```sql
SET STATEMENT sql_mode='maxdb' FOR
ALTER TABLE mysql.innodb_table_stats MODIFY last_update TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP() ON UPDATE CURRENT_TIMESTAMP();
CREATE TABLE t (a INT KEY)Engine=InnoDB STATS_PERSISTENT=1;
```